chainer/gradient_check.py

```diff
@@ -287,76 +278,74 @@ def check_backward(func, x_data, y_grad, params=(),
 
     for skip, x in six.moves.zip(no_grads, xs):
         if skip:
-            assert x.grad is None
+            if x.grad is not None:
+                raise RuntimeError(
+                    'gradient of int variable must be None')
         else:
             if x.grad is None:
                 raise RuntimeError(
                     'gradients of some arguments are not calculated')
 
+    if len(xs) - no_grads.count(True) + len(params) == 0:
+        # When there is no float variables, we need not to check gradient
+        # values
+        return
+
+    variables = _filter_list(xs, no_grads) + list(params)
     # Keep the gradient arrays of params which may be overwritten by func
-    params_grad = [param.grad for param in params]
-    param_data = [p.data for p in params]
-    casted_data = [x.data.copy() for x in casted_xs]
+    grads = [x.grad for x in variables]
```

Comment:
Is keeping the references of `x.grad` enough if they are overwritten? I suppose that overwritten are the values of `x.grad`.